wallet/src/main/scala/org/bitcoins/wallet/internal/FundTransactionHandling.scala

```diff
@@ -75,18 +74,8 @@ trait FundTransactionHandling extends WalletLogger { self: Wallet =>
         }
 
         // Need to remove immature coinbase inputs
-        coinbaseUtxos = utxoWithTxs.filter(_._2.isCoinbase)
-        confFs = coinbaseUtxos.map(utxo =>
-          chainQueryApi
-            .getNumberOfConfirmations(utxo._1.blockHash.get)
-            .map((utxo, _)))
-        confs <- FutureUtil.collect(confFs)
-        immatureCoinbases =
-          confs
-            .filter { case (_, confsOpt) =>
-              confsOpt.isDefined && confsOpt.get < Consensus.coinbaseMaturity
-            }
-            .map(_._1)
+        immatureCoinbases = utxoWithTxs.filter(
```

Comment:
:thinking_face: why could this be simplified so much? 

I guess i am more confused why we were doing things the way we were before? The new code seems correct

Reply:
The stuff here before was added before we had the immature coinbase txo state